src/components/WalletSelector.ts

```diff
@@ -0,0 +1,199 @@
+import { setupKeypom } from '@keypom/selector';
+import type { FinalExecutionOutcome, WalletSelector, WalletSelectorState } from '@near-wallet-selector/core';
+import { setupWalletSelector } from '@near-wallet-selector/core';
+import { setupHereWallet } from '@near-wallet-selector/here-wallet';
+import { setupLedger } from '@near-wallet-selector/ledger';
+import { setupMeteorWallet } from '@near-wallet-selector/meteor-wallet';
+import { setupMintbaseWallet } from '@near-wallet-selector/mintbase-wallet';
+import { setupModal } from '@near-wallet-selector/modal-ui';
+import { setupMyNearWallet } from '@near-wallet-selector/my-near-wallet';
+import { setupNearMobileWallet } from '@near-wallet-selector/near-mobile-wallet';
+import { setupNeth } from '@near-wallet-selector/neth';
+import { setupNightly } from '@near-wallet-selector/nightly';
+import { setupSender } from '@near-wallet-selector/sender';
+import { setupWelldoneWallet } from '@near-wallet-selector/welldone-wallet';
+import { providers, utils } from 'near-api-js';
+import { setupFastAuthWallet } from 'near-fastauth-wallet';
+import type { Context } from 'react';
+import { createContext } from 'react';
+import { distinctUntilChanged, map } from 'rxjs';
```

Comment:
Hey @gagdiez, if our end goal is to keep improving performance of `dev.near.org`, it makes me wonder if a heavy dependency like `rxjs` is worth it just for a couple of convenience methods. Would it be worth refactoring a bit to remove the dependency?

Reply:
all in for less dependencies

I used this because is what they recommend on wallet selector, and to be honest I have no idea what is happening

any chance you guide me on how to improve this one?